### PR TITLE
Update to Go 1.21

### LINKED ISSUE #34: Go 1.21 update check list
Once we drop support for 1.20, take a look at nspcc-dev/neo-go#3089.

### PR DESCRIPTION
**Pull request by smallhive**
Close #34

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -12,7 +12,8 @@ run:
 # output configuration options
 output:
   # colored-line-number|line-number|json|tab|checkstyle|code-climate, default is "colored-line-number"
-  format: tab
+  formats:
```
Comment (roman-khimov) [inline .golangci.yaml:15]: Please at least reference similar PRs in other repos. Or better yet apply commits from @cthulhu-rider (like in https://github.com/nspcc-dev/rfc6979/pull/8), they have perfect descriptions.

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -77,7 +77,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        go_versions: [ '1.20', '1.21', '1.22' ]
+        go_versions: [ '1.21', '1.22', '1.23' ]
```
Comment (roman-khimov) [inline .github/workflows/run_tests.yml:80]: 1.23 is a separate change, but we need to adapt to our new policy and drop one version from the set.

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.22-alpine as basebuilder
+FROM golang:1.23-alpine as basebuilder
```
Comment (roman-khimov) [inline Dockerfile:1]: Not here, it's a part of #43.

**Review by roman-khimov — COMMENTED** [review]
Looks like there is nothing to improve here code-wise.

File: .golangci.yaml (hunk 1 above)
Comment (smallhive) [reply .golangci.yaml:15]: Updated

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (smallhive) [reply .github/workflows/run_tests.yml]: Removed 1.23 and reduced to just couple versions

File: Dockerfile (hunk 3 above)
Comment (smallhive) [reply Dockerfile]: Removed from PR

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**